packages/wallet/ui/src/util/SuggestChain.js

```diff
@@ -5,36 +5,33 @@ export const AGORIC_COIN_TYPE = 564;
 export const COSMOS_COIN_TYPE = 118;
 
 /**
- * @param {string} networkConfig URL
- * @param {object} addrs
- * @param {string} addrs.rpcAddr URL or origin
- * @param {string} addrs.apiAddr URL
- * @param {string} chainId
- * @param {string} [caption]
+ * @param {object} io
+ * @param {typeof Math.random} io.random
+ * @param {import('@agoric/internal').NetworkConfig} networkConfig
+ * @param {string} caption
  * @returns {import('@keplr-wallet/types').ChainInfo}
  */
-const makeChainInfo = (
-  networkConfig,
-  { rpcAddr, apiAddr },
-  chainId,
-  caption,
-) => {
-  const coinType = Number(
-    new URL(networkConfig).searchParams.get('coinType') || AGORIC_COIN_TYPE,
-  );
-  const hostname = new URL(networkConfig).hostname;
-  const network = hostname.split('.')[0];
+export const makeChainInfo = (io, networkConfig, caption) => {
+  const { chainName, rpcAddrs, apiAddrs } = networkConfig;
+  const index = Math.floor(io.random() * rpcAddrs.length);
```

Comment:
passing in the random number generator for 1 call seems a little odd -- why not just pass in the results of the call?

Reply:
agreed. the dan voice in my head said, "pass the ocap" but that Dan is an crinkled facsimile of the real one